ISearchProvider interface is using derived class as parameter type in NotebookSearchProvider

When I use `@jupyterlab/documentsearch` in my custom extension and implement my own Search Provider, I am running into a build issue. Built-in search provider `NotebookSearchProvider`'s type definitions are causing the errors. Message is below.

```
node_modules/@jupyterlab/documentsearch/lib/providers/notebooksearchprovider.d.ts:11:5 - error TS2416: Property 'getInitialQuery' in type 'NotebookSearchProvider' is not assignable to the same property in base type 'ISearchProvider'.
  Type '(searchTarget: NotebookPanel) => any' is not assignable to type '(searchTarget: Widget) => any'.
    Types of parameters 'searchTarget' and 'searchTarget' are incompatible.
      Type 'Widget' is missing the following properties from type 'NotebookPanel': activated, session, content, model, and 25 more.

node_modules/@jupyterlab/documentsearch/lib/providers/notebooksearchprovider.d.ts:21:5 - error TS2416: Property 'startQuery' in type 'NotebookSearchProvider' is not assignable to the same property in base type 'ISearchProvider'.
  Type '(query: RegExp, searchTarget: NotebookPanel) => Promise<ISearchMatch[]>' is not assignable to type '(query: RegExp, searchTarget: Widget) => Promise<ISearchMatch[]>'.
    Types of parameters 'searchTarget' and 'searchTarget' are incompatible.
      Type 'Widget' is not assignable to type 'NotebookPanel'.
```

[Link to source](https://github.com/jupyterlab/jupyterlab/blob/master/packages/documentsearch/src/providers/notebooksearchprovider.ts#L26)

I think NotebookSearchProvider shouldn't change the signature of these interface methods and use the base type Widget as the searchTarget's type. That fixes the problem and CodeMirrorSearchProvider is doing that way as well.

Interestingly this builds fine with JupyterLab, only an issue when used in external extensions.

--- a/packages/documentsearch/src/providers/notebooksearchprovider.ts
+++ b/packages/documentsearch/src/providers/notebooksearchprovider.ts
@@ -8,6 +8,7 @@ import { CodeMirrorEditor } from '@jupyterlab/codemirror';
 import { Cell, MarkdownCell } from '@jupyterlab/cells';
 
 import { Signal, ISignal } from '@phosphor/signaling';
+import { Widget } from '@phosphor/widgets';
 
 import CodeMirror from 'codemirror';
 
@@ -23,8 +24,9 @@ export class NotebookSearchProvider implements ISearchProvider {
    *
    * @returns Initial value used to populate the search box.
    */
-  getInitialQuery(searchTarget: NotebookPanel): any {
-    const activeCell = searchTarget.content.activeCell;
+  getInitialQuery(searchTarget: Widget): any {
+    const notebookPanel = searchTarget as NotebookPanel;
+    const activeCell = notebookPanel.content.activeCell;
     const selection = (activeCell.editor as CodeMirrorEditor).doc.getSelection();
     // if there are newlines, just return empty string
     return selection.search(/\r?\n|\r/g) === -1 ? selection : '';
@@ -41,9 +43,9 @@ export class NotebookSearchProvider implements ISearchProvider {
    */
   async startQuery(
     query: RegExp,
-    searchTarget: NotebookPanel
+    searchTarget: Widget
   ): Promise<ISearchMatch[]> {
-    this._searchTarget = searchTarget;
+    this._searchTarget = searchTarget as NotebookPanel;
     const cells = this._searchTarget.content.widgets;
 
     this._query = query;

